src/Models/Revision.php

```diff
@@ -188,55 +231,64 @@ public function isUpdatedAt(Checkpoint $moment): bool
     }
 
     /**
-     * Return all the revisions sibling that share the same item
+     * Return all the revisions sibling that share the same item except the current one
      *
-     * @return
+     * @return HasMany
      */
     public function otherRevisions(): HasMany
     {
-        return $this->allRevisions()->where('id', '!=', $this->id);
+        return $this->allRevisions()->where('id', '!=', $this->getKey());
     }
 
     /**
+     * Retrieve the latest revision ids within the boundary window given valid checkpoints, carbon or datetime strings
+     *
      * @param  Builder  $q
-     * @param  Checkpoint|Carbon|string|null  $until
-     * @param  Checkpoint|Carbon|string|null  $since
+     * @param  Checkpoint|\Illuminate\Support\Carbon|string|null  $until  valid checkpoint, carbon or datetime string
+     * @param  Checkpoint|\Illuminate\Support\Carbon|string|null  $since  valid checkpoint, carbon or datetime string
      * @return Builder
      */
     public function scopeLatestIds(Builder $q, $until = null, $since = null)
     {
-        $q->withoutGlobalScopes()->selectRaw("max({$this->getQualifiedKeyName()}) as closest")
-            ->groupBy(['original_revisionable_id', 'revisionable_type'])->orderBy(DB::raw('NULL'));
-
+        $q->withoutGlobalScopes()->selectRaw("max({$this->getKeyName()}) as closest")
+            ->groupBy(['original_revisionable_id', 'revisionable_type'])->orderByDesc('previous_revision_id');
 
         $checkpoint = config('checkpoint.checkpoint_model', Checkpoint::class);
-        $checkpointDateColumn = $checkpoint::CHECKPOINT_DATE;
 
         if ($until instanceof $checkpoint) {
-            // where in this checkpoint or one of the previous ones
-            $q->whereIn(
-                $this->getCheckpointIdColumn(),
-                $checkpoint::select('id')->where($checkpointDateColumn, '<=', $until->$checkpointDateColumn)
-            );
+            // where in given checkpoint or one of the previous ones
+            $q->whereIn($this->getCheckpointIdColumn(), $checkpoint::olderThanEquals($until)->select('id'));
         } elseif ($until !== null) {
-            $q->where($this->getQualifiedCreatedAtColumn(), '<=', Carbon::parse($until));
+            $q->where($this->getQualifiedCreatedAtColumn(), '<=', $until);
         }
 
         if ($since instanceof $checkpoint) {
-            // where in this checkpoint or one of the following ones
-            $q->whereIn(
-                $this->getCheckpointIdColumn(),
-                $checkpoint::select('id')->where($checkpointDateColumn, '>', $since->$checkpointDateColumn)
-            );
+            // where in one of the newer checkpoints than given
+            $q->whereIn($this->getCheckpointIdColumn(), $checkpoint::newerThan($since)->select('id'));
         } elseif ($since !== null) {
-            $q->where($this->getQualifiedCreatedAtColumn(), '>', Carbon::parse($since));
```

Comment:
Doesn't removing this mean we lose the ability to pass in strings like "tomorrow"? i know that probably wont come up super often, but seems like it could be handy, no?

Reply:
I removed it to avoid the errors Carbon::parse throws when it receives strings that don't conform to `strtotime()`, ie the checkpoint slugs. Besides I think in this case, less is more. We already provide 3 ways to interpret these variables and a user could just run `Revisionable::at(strtotime("tomorrow"))->get()` to achieve the same result.

Reply:
Yeah good point there, no problem!